packages/dev/core/src/types.ts

```diff
@@ -27,6 +142,29 @@ export type IndicesArray = number[] | Int32Array | Uint32Array | Uint16Array;
  */
 export type DataArray = number[] | ArrayBuffer | ArrayBufferView;
 
+/**
+ * A recursive array of T (i.e. T[], T[][], T[][][], ...)
+ */
+export type RecrusiveArray<T> = (RecrusiveArray<T> | T)[];

```

Comment:
Let s remove if unused ?

Reply:
This will almost certainly be used in future parts of the Tensor API, and provides a very helpful type for end users. It was intended to also be used for the Tensor type parameter, but could not do so because of issues with Typescript.